exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/metrics_encoder/__init__.py

```diff
@@ -272,18 +278,18 @@ def encode_metrics(data: MetricsData) -> ExportMetricsServiceRequest:
                 elif isinstance(metric.data, ExponentialHistogramType):
                     for data_point in metric.data.data_points:
 
-                        if data_point.positive.bucket_counts:
+                        if positive_buckets := truncate_trailing_zeros(data_point.positive.bucket_counts):
```

Comment:
maybe move truncation logic into `def bucket_counts` implementation to avoid duplication?